Spigot-Server-Patches/0555-Zero-pad-chunks-before-saving-them-to-the-disk.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Toon Schoenmakers <nighteyes1993@gmail.com>
+Date: Mon, 24 Aug 2020 15:12:06 +0200
+Subject: [PATCH] Zero pad chunks before saving them to the disk
+
+This makes the unused data inside region files more repeatable.
+Which will result in better compression in case the entire world directory
+is compressed (like in backups). Which will naturally result in smaller backups.
+
+diff --git a/src/main/java/net/minecraft/server/RegionFile.java b/src/main/java/net/minecraft/server/RegionFile.java
+index 939b51826681a7648ad3ea2512c20082193e6a01..2ca18af9342702bdb246eb99b8a46f3ee1ec53a7 100644
+--- a/src/main/java/net/minecraft/server/RegionFile.java
++++ b/src/main/java/net/minecraft/server/RegionFile.java
+@@ -517,6 +517,13 @@ public class RegionFile implements AutoCloseable {
+             ByteBuffer bytebuffer = ByteBuffer.wrap(this.buf, 0, this.count);
+ 
+             bytebuffer.putInt(0, this.count - 5 + 1);
++
++            // Paper start
++            int remaining = this.count % 4096;
```

Comment:
this is better served by having a single reused buffer of the entire block size, calculate the last block the "partial" data writes into, and then zero out that block and the remainder BEFORE writing the real data, so we can avoid allocating buffers every write.

This static zero buffer should also be a Direct allocated byte buffer too.

Reply:
This should also use & 4095 instead. Mod operator is slower than bit and.